cmd/dep/init.go

```diff
@@ -239,6 +242,45 @@ func hasImportPathPrefix(s, prefix string) bool {
 	return strings.HasPrefix(s, prefix+"/")
 }
 
+// itemizedFeedback itemizes resolved dependencies and logs the solution as
+// feedback.
+func itemizedFeedback(m *dep.Manifest, l *dep.Lock, pkgT pkgtree.PackageTree) {
+	// Algo:
+	// loop through lock projects
+	// 1. check in manifest for for same entry
+	// 2. if exists in manifest, it's direct dep, goes to manifest & lock
+	// 3. if not exists in manifest, check in pkgT
+	//	3.a. if exists in pkgT, it's direct dep, hint constraint, lock only
+	//	3.b. if not exists in pkgT, it's transitive dep, lock only
+	// 4. log feedback
```

Comment:
Is this a good algo to separate direct and transitive deps? Any other simpler solution?